src/options.rs

```diff
@@ -0,0 +1,284 @@
+            return
+        }
+
+        // e.g. `#[setters]`
+        if let syn::MetaItem::Word(ref ident) = attr.value {
+            match ident.as_ref() {
+                "setters" => {
+                        self.setter_enabled(true);
+                        return
+                    },
+                "getters" => {
+                        self.getter_enabled(true);
+                        return
+                    },
+                _ => {}
+            }
+        }
+
+        // e.g. `#[setters(...)]`
+        if let syn::MetaItem::List(ref ident, ref nested_attrs) = attr.value {
+            match ident.as_ref() {
+                "setters" => {
+                        self.setter_enabled(true);
+                        self.parse_setters_options(nested_attrs);
+                        return
+                    },
+                "getters" => {
+                        self.getter_enabled(true);
+                        self.parse_getters_options(nested_attrs);
+                        return
+                    },
+                _ => {}
+            }
+        }
+        debug!("Ignoring attribute.");
+    }
+
+    fn parse_setters_options(&mut self, nested: &[syn::NestedMetaItem]) {
+        trace!("Parsing setter options.");
+        for x in nested {
+            if let syn::NestedMetaItem::MetaItem(ref meta_item) = *x {
+                self.parse_setters_options_metaItem(meta_item)
+            } else {
+                panic!("Expected NestedMetaItem::MetaItem, found {:?}", x)
+            }
+        }
+    }
+
+    #[allow(non_snake_case)]
+    fn parse_setters_options_metaItem(&mut self, meta_item: &syn::MetaItem) {
+        trace!("Parsing MetaItem {:?}", meta_item);
+        match *meta_item {
+            syn::MetaItem::Word(ref ident) => {
+                self.parse_setters_options_word(ident)
+            },
+            syn::MetaItem::NameValue(ref ident, ref lit) => {
+                self.parse_setters_options_nameValue(ident, lit)
+            },
+            _ => {
+                panic!("Expected MetaItem::Word/NameValue, found {:?}", meta_item)
+            }
+        }
+    }
+
+    /// e.g `owned` in `#[setters(owned)]`
+    fn parse_setters_options_word(&mut self, ident: &syn::Ident) {
+        trace!("Parsing word {:?}", ident);
+        match ident.as_ref() {
+            "owned" => {
+                self.setter_pattern(SetterPattern::Owned)
+            },
+            "mutable" => {
+                self.setter_pattern(SetterPattern::Mutable)
+            },
+            "immutable" => {
+                self.setter_pattern(SetterPattern::Immutable)
+            },
+            "public" => {
+                self.setter_public(true)
+            },
+            "private" => {
+                self.setter_public(false)
+            },
+            _ => {
+                panic!("Unknown option {:?}", ident)
+            }
+        };
```

Comment:
Yes, this kind of separation is, what I was looking for. :-)
Sounds interesting, I'll have a look at serde. Never used it before, but since everyone is talking about that's probably an education gap I should close anyway. :-)